src/app/core/services/label.service.ts

```diff
@@ -110,16 +114,27 @@ export class LabelService {
   }
 
   /**
-   * Converts the (color) hex value into RGB format
-   * @param hex: the hex value
+   * Determines if the color code provided is "dark" by calculating its luminance and
+   * comparing it with a constant threshold value
+   * @param inputColor: the color code input
+   * @return true if the luminance is less than the threshold (darker), false otherwise
    */
-  hexToRgb(hex: string) {
-    const rgbResult = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
-    return rgbResult ? {
-      r: parseInt(rgbResult[1], 16),
-      g: parseInt(rgbResult[2], 16),
-      b: parseInt(rgbResult[3], 16)
-    } : null;
+  isDarkColor(inputColor: string) {
+    const color = (inputColor.charAt(0) === '#') ? inputColor.substring(1, 7) : inputColor;
+    const r = parseInt(color.substring(0, 2), 16);
+    const g = parseInt(color.substring(2, 4), 16);
+    const b = parseInt(color.substring(4, 6), 16);
+    const uiColors = [r / 255, g / 255, b / 255];
+    const c = uiColors.map((col) => {
+      if (col <= 0.03928) {
+        return col / 12.92;
+      }
+      return Math.pow((col + 0.055) / 1.055, 2.4);
+    });
+    // Calculate the luminance of the color
+    const L = (0.2126 * c[0]) + (0.7152 * c[1]) + (0.0722 * c[2]);
+    // A higher threshold value will result in more colors determined to be "dark"
+    return (L < BGCOLOR_THRESHOLD);
   }
```

Comment:
Do we have some unit testing framework? These utility functions can be easily unit tested.

Reply:
Currently we do not have testing framework, but we can setup some testing framework in another PR

Reply:
Let's do that in a separate PR then.